vmm/guest/gpm.c

```diff
@@ -202,6 +203,17 @@ void gpm_set_mapping(IN guest_handle_t guest,
 		}
 	}
 
+	/* For mapping [top_of_memory, MAX_PHYS_ADDR), set cache type with Ucached */
+	if (size_tmp != 0) {
+		if ((gpa_tmp == hpa_tmp) &&
+		    (hpa_tmp >= top_of_memory) &&
+		    (hpa_tmp <= MAX_PHYS_ADDR) &&
+		    (hpa_tmp + size_tmp) <= MAX_PHYS_ADDR) {
```

Comment:
since it is a workaround, how about making it simpler by just checking if the "size" is "MAX_PHYS_ADDR" with some comments?

Reply:
OK, will refine it and add comments